src/main/java/org/gridsuite/explore/server/services/FilterService.java

```diff
@@ -6,139 +6,94 @@
 
-import org.gridsuite.explore.server.ExploreException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.core.ParameterizedTypeReference;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
+import org.springframework.http.*;
 import org.springframework.stereotype.Service;
-import org.springframework.web.reactive.function.client.ClientResponse;
-import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.client.RestTemplate;
 import org.springframework.web.util.UriComponentsBuilder;
-import reactor.core.publisher.Flux;
-import reactor.core.publisher.Mono;
-import reactor.core.scheduler.Schedulers;
 
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
-import java.util.logging.Level;
 import java.util.stream.Collectors;
 
-import static org.gridsuite.explore.server.ExploreException.Type.FILTER_NOT_FOUND;
-
 /**
  * @author Etienne Homer <etienne.homer at rte-france.com>
  */
 @Service
 public class FilterService implements IDirectoryElementsService {
-    private static final String ROOT_CATEGORY_REACTOR = "reactor.";
-
     private static final String FILTER_SERVER_API_VERSION = "v1";
 
     private static final String DELIMITER = "/";
 
-    private final WebClient webClient;
     private String filterServerBaseUri;
 
+    private final RestTemplate restTemplate;
+
     @Autowired
-    public FilterService(@Value("${backing-services.filter-server.base-uri:http://filter-server/}") String filterServerBaseUri,
-                         WebClient.Builder webClientBuilder) {
+    public FilterService(@Value("${backing-services.filter-server.base-uri:http://filter-server/}") String filterServerBaseUri, RestTemplate restTemplate) {
         this.filterServerBaseUri = filterServerBaseUri;
-        this.webClient = webClientBuilder.build();
+        this.restTemplate = restTemplate;
     }
 
     public void setFilterServerBaseUri(String filterServerBaseUri) {
         this.filterServerBaseUri = filterServerBaseUri;
     }
 
-    public Mono<Void> replaceFilterWithScript(UUID id) {
+    public void replaceFilterWithScript(UUID id) {
         String path = UriComponentsBuilder.fromPath(DELIMITER + FILTER_SERVER_API_VERSION + "/filters/{id}/replace-with-script")
                 .buildAndExpand(id)
                 .toUriString();
-
-        return webClient.put()
-                .uri(filterServerBaseUri + path)
-                .retrieve()
-                .onStatus(httpStatus -> httpStatus == HttpStatus.NOT_FOUND, clientResponse -> Mono.error(new ExploreException(FILTER_NOT_FOUND)))
-                .bodyToMono(Void.class)
-                .publishOn(Schedulers.boundedElastic())
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+        restTemplate.exchange(filterServerBaseUri + path, HttpMethod.PUT, null, Void.class);
```

Comment:
keep the following behaviour 
.onStatus(httpStatus -> httpStatus == HttpStatus.NOT_FOUND, clientResponse -> Mono.error(new ExploreException(FILTER_NOT_FOUND)))

Reply:
ok i will return the try catch bloc here